File: week2/Eren-Taskiran/567-permutation-in-string.java
class Solution {
    public boolean checkInclusion(String s1, String s2) {
        if(s1.length()>s2.length()){
           return false;
        } 
        char[] arr1 = s1.toCharArray();
        char[] arr2 = s2.toCharArray();
        int[] map = new int[26];
        int[] check = new int[26];
        int j = arr1.length;

        for(int i = 0; i<j;i++){
            check[arr1[i]-'a']++;
            map[arr2[i]-'a']++;
        }
        if(Arrays.equals(check, map)) {
            return true;
        }
        for(int i = j; i<arr2.length;i++){
            map[arr2[i-j]-'a']--;
            map[arr2[i]-'a']++;
            if(Arrays.equals(map,check)) return true;
        }

        
        return false;
    }
    
}

File: week2/atakantekatan/567. Permutation in String.py
class Solution:
    def checkInclusion(self, s1: str, s2: str) -> bool:
        if len(s1)>len(s2):
            return False

        c1 = Counter()
        cw = Counter()
        l, r = 0, len(s1)

        for i in range(l,r):
            c1[s1[i]] += 1
            cw[s2[i]] += 1

        while r < len(s2) and c1 != cw:
            cw[s2[l]] -= 1
            if cw[s2[l]] == 0:
                del cw[s2[l]]
            cw[s2[r]] += 1
            l += 1
            r += 1
            
        return True if c1 == cw else False